### PR TITLE
Nerfs biofoam, standardises pain-getting and adds a new painkiller

### PR DESCRIPTION
**Pull request by X0-11**
:cl: XO-11
tweak: Biofoam now leaves behind a chemical that will cause further injections to be nullified.
tweak: Fixes the cost-based system behind biofoam healing to actually function.
tweak: Makes biofoam heal-messages less spammy.
tweak: Most occurances of pain-getting now take painkillers into account.
tweak: Returns painkillers to their original values due to above change.
tweak: Modifies the lifetime of some painkillers.
tweak: Makes the blurry eyes effect a bit less obstructive.
rscadd: Adds a new painkiller: Polypseudomorphine. A very powerful painkiller intended for combat use. Can also work as a short term sedative if a dose exceeding one standard pill is taken.
/:cl:

### CODE REVIEW

File: code/modules/halo/chemicals/medicine.dm [hunk 1]
```diff
@@ -113,27 +135,37 @@
 
 /datum/reagent/biofoam/affect_blood(var/mob/living/carbon/M,var/alien,var/removed) //Biofoam stops internal and external bleeding, heals organs and fixes bones.
 	if(istype(M,/mob/living/carbon/human))
+		if(locate(/datum/reagent/biofoampenaliser) in holder.reagent_list)
```
Comment (bloxgate) [inline code/modules/halo/chemicals/medicine.dm]: Isn't this already done by a chemical reaction? Forgive me if I'm a bit out of date on the processing order, but I'd expect that reaction to happen before the reagents in blood are processed based on past behavior.

File: code/modules/halo/chemicals/medicine.dm [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
-#define BIOFOAM_COST_FIXWOUNDS 0.25
-#define BIOFOAM_COST_REMOVESHRAP 2
+#define BIOFOAM_COST_STOPBLEED 0.2
+#define BIOFOAM_COST_MENDBONE 0.3
+#define BIOFOAM_COST_MENDINTERNAL 0.1
+#define BIOFOAM_COST_MENDEXTERNAL 0.15
+#define BIOFOAM_COST_FIXWOUNDS 0.1
+#define BIOFOAM_COST_REMOVESHRAP 0.2
+
+#define POLYPSEUDOMORPHINE_PAINKILL 400
```
Comment (bloxgate) [inline code/modules/halo/chemicals/medicine.dm:8]: Are we sure we want to go ahead and replace tramadol and paracetamol with this in the medkits? This is nearly 5 times stronger than tramadol and twice as strong as the oxycodone autoinjectors in the combat medkits. I could understand maybe replacing those autoinjectors, but all of the tramadol and paracetamol bottles? That seems to be a bit excessive given how much more effective this is.

If pain-crit is still a major issue, maybe we should consider bumping up the numbers on the other medicines or changing how CE_PAINKILLER is applied instead?

File: code/modules/halo/chemicals/medicine.dm (hunk 1 above)
Comment (X0-11) [reply code/modules/halo/chemicals/medicine.dm]: I think it is, yeah. This was added during testing and I forgot to remove it

File: code/modules/halo/chemicals/medicine.dm (hunk 2 above)
Comment (X0-11) [reply code/modules/halo/chemicals/medicine.dm:8]: tramadol and paracetamol aren't really "I have taken a bullet wound" type medicines, although they can help dampen the pain of a hit as they last a lot longer than oxy or polypseudo

**Review by BDpuffy420 — APPROVED** [review]

**Pull request merged**